cogs/live_ping.py

```diff
@@ -0,0 +1,470 @@
+from __future__ import annotations
+
+import asyncio
+import datetime
+import logging
+import re
+from typing import TYPE_CHECKING, Literal, TypedDict
+from urllib.parse import quote_plus
+
+import aiohttp
+import discord
+import Paginator
+import yarl
+from discord.ext import commands, tasks
+
+from config import LIVE_SESSION_CALENDARS, LIVE_SESSION_PING_ROLE
+from utils.formats import plural
+
+
+if TYPE_CHECKING:
+    from typing_extensions import NotRequired
+
+    from bot import IITMBot
+
+
+DATE_FORMAT = "%d-%m-%Y %H:%M"
+REMINDER_BEFORE_N_MINUTES = 5
+MEET_LINK_REGEX = re.compile(r"(https?://meet\.google\.com/[a-zA-Z0-9_-]+)")
+
+log = logging.getLogger(__name__)
+
+
+class EventPayload(TypedDict):
+    id: str
+    summary: NotRequired[str]
+    description: NotRequired[str]
+    start: _DateTimePayload
+    end: _DateTimePayload
+    conferenceData: _ConferenceData
+
+
+class _DateTimePayload(TypedDict):
+    dateTime: str
+    timeZone: str
+
+
+class _ConferenceData(TypedDict):
+    conferenceId: str
+    conferenceSolution: _ConferenceSolution
+
+
+class _ConferenceSolution(TypedDict):
+    key: _ConferenceSolutionKey
+
+
+class _ConferenceSolutionKey(TypedDict):
+    type: Literal["eventHangout", "eventNamedHangout", "hangoutsMeet", "addOn"]
+
+
+class Event:
+    def __init__(self, event: EventPayload) -> None:
+        self.name: str = event.get("summary", "No Name")
+        self.desc: str = event.get("description", "No Description")
+        self.start: datetime.datetime = datetime.datetime.fromisoformat(event["start"]["dateTime"]).astimezone()
+        self.end: datetime.datetime = datetime.datetime.fromisoformat(event["end"]["dateTime"]).astimezone()
+        self.tz: str = event["start"]["timeZone"]
+        self.id: str = event["id"]
+        cData = event.get("conferenceData", None)
+        if not cData:
+            self.meet_link = (
+                None
+                if (meet_links := extract_google_meet_links(self.desc)) is None or len(meet_links) == 0
+                else meet_links[0]
+            )
```

Comment:
This is unreadable as all hell, and also the walrus is evil.
```suggestion
            match = MEET_LINK_REGEX.search(self.desc)
            if match is not None:
                self.meet_link = match.group()
            else:
                self.meet_link = None
```